client/src/components/articles/ArticleLayout.vue

```diff
@@ -0,0 +1,102 @@
+    <SingleArticle :article="getSingleArticle(Number(route.params.id))!" />
+    <button
+      v-if="getUserId"
+      id="likeButton"
+      class="flex h-16 w-28 rounded-xl transition-transform duration-300 ease-in-out hover:scale-125 hover:cursor-pointer phone:ml-2 phone:h-14 phone:w-18"
+      :class="getLikedClass"
+      @click.prevent="likeArticle"
+    >
+      <LikeIcon />
+      <p class="ml-1 mt-2">{{ getSingleArticle(Number(route.params.id))!.likes }}</p>
+    </button>
+    <div class="mt-12 h-fit w-full rounded-xl bg-white p-2">
+      <SortComment v-if="rootComments.length"/>
+      <button
+        v-if="!isCreateCommentVisible && getUserId"
+        class="mb-6 mt-2 h-12 w-32 bg-gray-300 text-center text-lg text-black hover:cursor-pointer"
+        @click="toggleCreateComment"
+      >
+        Comment
+      </button>
+      <CreateComment
+        v-if="isCreateCommentVisible"
+        comment="Create"
+        :user="user"
+        @cancelComment="toggleCreateComment"
+      />
+      <div v-for="(articleComment, index) in rootComments" :key="index">
+        <ArticleComment
+          :user="user"
+          :comment="articleComment"
+        />
+        <div v-for="(reply, index) in articleComment.replies" :key="index">
+          <div class="ml-40 phone:ml-6">
+            <ArticleComment
+              v-if="articleComment.replies.length"
+              :user="user"
+              :comment="reply"
+            />
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, computed, onMounted } from 'vue'
+import { RouterLink, useRoute } from 'vue-router'
+import { storeToRefs } from 'pinia'
+import { useArticleStore } from '../../stores/ArticleStore'
+import { useCommentStore } from '../../stores/CommentStore'
+import { useUserStore } from '../../stores/UserStore'
+import SingleArticle from './SingleArticle.vue'
+import ArticleComment from './ArticleComment.vue'
+import CreateComment from './CreateComment.vue'
+import SortComment from './SortComment.vue'
+import TitleSeparator from '../baseComponents/TitleSeparator.vue'
+import LikeIcon from '../icons/LikeIcon.vue'
+
+import { Comment } from '../../types/comment.ts'
+
+const route = useRoute()
+const { likedArticle, viewedArticle } = useArticleStore()
+const { getSingleArticle } = storeToRefs(useArticleStore())
+const { getSortedComments } = storeToRefs(useCommentStore())
+const { user, getUserId } = storeToRefs(useUserStore())
+
+
+const rootComments = computed(() => {
+  return getSortedComments.value(Number(route.params.id)).filter((comment: Comment) => !comment.parentId)
+})
+
+const isCreateCommentVisible = ref(false)
+const isLiked = computed(() => {
+  return (getSingleArticle.value(Number(route.params.id))!.likedBy.includes(getUserId.value))
```

Comment:
Create a computed from Number(route.params.id) so you dont have to call the Number() function multiple times.